src/libraries/System.Net.Sockets/src/System/Net/Sockets/SendPacketsElement.cs

```diff
@@ -102,14 +102,24 @@ public SendPacketsElement(byte[] buffer, int offset, int count, bool endOfPacket
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            Initialize(null, null, buffer, offset, count, endOfPacket);
+            Initialize(null, null, buffer, new ReadOnlyMemory<byte>(buffer, offset, count), offset, count, endOfPacket);
         }
 
-        private void Initialize(string? filePath, FileStream? fileStream, byte[]? buffer, long offset, int count, bool endOfPacket)
+        public SendPacketsElement(ReadOnlyMemory<byte> buffer) :
+            this(buffer, false)
+        { }
+
+        public SendPacketsElement(ReadOnlyMemory<byte> buffer, bool endOfPacket)
+        {
+            Initialize(null, null, null, buffer, 0, buffer.Length, endOfPacket);
+        }
+
+        private void Initialize(string? filePath, FileStream? fileStream, byte[]? buffer, ReadOnlyMemory<byte>? memoryBuffer, long offset, int count, bool endOfPacket)
         {
             FilePath = filePath;
             FileStream = fileStream;
             Buffer = buffer;
+            MemoryBuffer = memoryBuffer;
```

Comment:
Rather than storing buffer/offset/count and also storing a `ReadOnlyMemory<byte>`, did you consider just storing the `ReadOnlyMemory<byte>`?  If we need the `byte[]` for something (such as if someone accesses the getter), we can use MemoryMarshal.TryGetArray to extract the buffer/offset/count.

Reply:
It seems slightly weird to me that if you construct it with a `ReadOnlyMemory<byte>`, that you'd be able to get the byte[]/offset/count out of it via the existing properties. I know you can always do this with MemoryMarshal, but that's sort of an advanced API and most people won't ever deal with it.

Note we have to keep the offset/count fields for file cases.

All that said, I don't feel particularly strongly about this.

Reply:
You could store whether it came from a byte[] in a bool and only give back the array if it did. 

But my motivation here was avoiding increasing the size of the type. If the two ints are needed,  regardless,  meh.

Reply:
Ok, I'll keep as is.